src/main/java/net/solmis/cityparking/Ticket.java

```diff
@@ -0,0 +1,70 @@
+package net.solmis.cityparking;
+
+import java.time.ZonedDateTime;
+
+public class Ticket {
+    private long id;
+    private VehicleId vehicleId;
+    private ZonedDateTime startTimestamp;
+    private ZonedDateTime endTimestamp;
+
+    private Ticket(VehicleId vehicleId) {
+        this.vehicleId = vehicleId;
+    }
+
+    public static Ticket createTicketAndSetStartTimestamp(VehicleId vehicleId) {
+        Ticket newTicket = new Ticket(vehicleId);
+        newTicket.startTimestamp = ZonedDateTime.now();
+        return newTicket;
+    }
+
+    public static Ticket get(long id) {
+        return FakeDatabaseService.getInstance().getTicket(id);
+    }
+
+    public static boolean existsFor(VehicleId vehicleId) {
+        return FakeDatabaseService.getInstance().existsActiveTicketForVehicleId(vehicleId);
+    }
+
+    public void setEndTimestamp() {
+        this.endTimestamp = ZonedDateTime.now();
+    }
+
+    public void save() {
+        if (isRecord())
+            FakeDatabaseService.getInstance().updateTicket(this);
```

Comment:
niby wygląda fajnie, ale jak tu podłożymy prawdziwą bazę danych?

Reply:
Nie jestem niestety jeszcze za bardzo zaznajomiony z konceptem mockowania bazy danych. Starałem się ją jakoś zaimitować, by możliwe było przetestowanie tego API. Istotnie nie jest łatwo zaadaptować to rozwiązanie dla innej bazy danych. Lepiej by było tu dać odwołanie np. do DatabaseService, który by mógł w zależności od konfiguracji korzystać z fikcyjnej bazy (np. załadowanej z jsona) lub z prawdziwej bazy danych (np. SQLowej).